metron-interface/metron-alerts/src/app/pcap/pcap-filters/pcap-filters.component.ts

```diff
@@ -63,9 +63,13 @@ export class PcapFiltersComponent implements OnInit, OnChanges {
     this.model.endTimeMs = new Date(this.endTimeStr).getTime();
     if (this.ipSrcPort !== '') {
       this.model.ipSrcPort = +this.ipSrcPort;
+    } else {
```

Comment:
This is probably a dumb question, but why do we have to specifically delete the value?  Don't we treat empty string the same as missing?  If that's the case, why isn't this just 
```
this.model.ipSrcPort = +this.ipSrcPort;
this.model.ipDstPort = +this.ipDstPort;
```

Sidenote, are the pluses in here even doing anything? Could it just be
```
this.model.ipSrcPort = this.ipSrcPort;
this.model.ipDstPort = this.ipDstPort;
```

Reply:
They are different types, one is a string and one is a number.  I believe we created the this.ipSrcPort and this.ipDstPort string variables for regex validation purposes.  As such we can't just assign an empty string to a number type.  The + operator converts a string to a number.